docs: Deploy Loki Helm on AWS guide (#14517)
grafana/loki: docs/sources/setup/install/helm/install-microservices/_index.md

---
title: Loki Microservice Helm chart 
menuTitle: Install microservice Loki
description: Installing Loki in microservice mode using the Helm chart.
weight: 300
keywords: 
---

> # Loki Microservice Helm chart

Reviewer: This heading needs to revert back to the original to 1) match our Style Guide and 2) match the other installation headings (see the generated TOC to see what I mean).

- **QueryFrontend component** (2 replicas, maxUnavailable: 1): Manages frontend queries. Up to 1 replica can be unavailable during updates.
- **QueryScheduler component** (2 replicas): Schedules queries.

{{< admonition type="note" >}}

> We do not recommended to run Microservice mode with `filesystem` storage. For the purpose of this guide, we will use MinIO as the object storage to provide a complete example. 

Reviewer: ```suggestion
We do not recommend running Microservice mode with `filesystem` storage. For the purpose of this guide, we will use MinIO as the object storage to provide a complete example. 
```

   helm repo add grafana https://grafana.github.io/helm-charts
   ```

1. Update the chart repository:

   ```bash
   helm repo update
   ```

1. Create the configuration file `values.yaml`. The example below illustrates how to deploy Loki in test mode using MinIO as storage:

     ```yaml
     loki:
        schemaConfig:
          configs:

>             - from: 2024-04-01

Reviewer: ```suggestion
            - from: "2024-04-01"
```

   helm repo add grafana https://grafana.github.io/helm-charts
   ```

1. Update the chart repository:

   ```bash
   helm repo update
   ```

1. Create the configuration file `values.yaml`. The example below illustrates how to deploy Loki in test mode using MinIO as storage:

     ```yaml
     loki:
        schemaConfig:
          configs:
            - from: "2024-04-01"
              store: tsdb

>               object_store: s3

Author: So since MinIO is S3 compatible storage we leave the object storage type as s3.

              schema: v13
              index:
                prefix: loki_index_
                period: 24h
        ingester:
          chunk_encoding: snappy
        querier:
          # Default is 4, if you have enough memory and CPU you can increase, reduce if OOMing

>           max_concurrent: 4

Author: So max_concurrent is how many queries each querier can run at one time vs the number.

so in this case each querier can run a max of 4 queries concurrently at one time

  replicas: 0

```
{{< /collapse >}}

To configure other storage providers, refer to the [Helm Chart Reference]({{< relref "../reference" >}}).

## Deploying the Loki Helm chart to a Production Environment

{{< admonition type="note" >}}
We are actively working on providing more guides for deploying Loki in production. 
{{< /admonition >}}

> It is recommended to run Loki at scale within in a cloud environment like AWS, Azure, or GCP. The below guides will show you how to deploy a minimally viable production environment.

Reviewer: ```suggestion
It is recommended to run Loki at scale within a cloud environment like AWS, Azure, or GCP. The below guides will show you how to deploy a minimally viable production environment.
```
